Kennric/pelican howto

Kennric opened the pull request:
Add instructions for updating pelican sites

ramereth commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,406 @@
+What is Pelican
+---------------
+
+Pelican is a static site generator. In short, it allows us to create a 
```
Can you please add a link to the Pelican website here?

ramereth commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,406 @@
+client you use. For examples in this document, I'll use the git 
+command-line client, which can be used on all platforms. 
+
+.. code-block:: none
```
If you're not setting what type of code block this is, you can just use ``::`` instead. Can you please do that for all instances using ``none`` please?

ramereth commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,406 @@
+	git checkout -b your_name/a_descriptive_branch_name
+
+
+The string "your_name/a_descriptive_branch_name" is the full name of the 
```
Make ``your_name/a_descriptive_branch_name`` a tech block.

ramereth commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,406 @@
+The next step is to have your changes approved by the page owner. If you are 
+the page owner, it's a very good idea to have someone else look over your
+changes. You can 'tag' other users in the 'Conversation' tab of the pull 
+request by adding an @ to their username in a comment:
```
It might be good to create a group (similar to what we have as ``@chefs``) as a standard review user tag.

ramereth commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,406 @@
+
+.. code-block:: none
+
+	My changes are complete, @Ramereth, can you review?
```
Its better to just do ``@ramereth please review my changes``.

ramereth commented:
@Kennric A few other things to note:

* [ ] There are a lot of extra tailing white spaces, can you please clean those up?
* [ ] There are a lot of extra new lines between sections. We usually only have one new line between each section.
* [ ] Are we going to add a section which talks about how you can build the site for advanced users?

ramereth commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,406 @@
+Merge Into Staging
+------------------
+
+If you are the page owner, please read the `Github documentation on merging pull requests <https://help.github.com/articles/merging-a-pull-request/>`_
```
Can you please convert this to referencing the URL later using ``.. _foo: URL`` so that it doesn't extend longer than 80char?

ramereth replied:
Can you please convert this to referencing the URL later using ``.. _foo: URL`` so that it doesn't extend longer than 80char?

Kennric commented:
@ramereth  @kelnera Please review changes, I've added and cleaned up a lot.

ramereth - I think building the site locally is for another document, this is mostly aimed at editors, and already feels over-loaded.

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+==============================
+
+This documentation applies to both the CASS main website, the OSL website and 
+any other sites using the Pelican static site generator.
```
I would consider linking to a webpage explaining what SSG's are here. If this is new information for the reader it's good to put it in a hyperlink.

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+Git is a system that tracks changes to files. It allows multiple people to edit 
+the same set of files and merge their changes together. All the files, and a 
+history of all the changes made to them, are stored in such a way that you can 
+retrieve the past state of any file easily.
```
I like to explain what problem git solves, like "*git solves the multiple revisions problem seen in [this webcomic](http://www.phdcomics.com/comics/archive.php?comicid=1531)*"

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+Git via GitHub
+~~~~~~~~~~~~~~
+
+You can do most git operations entirely from the Github web UI. If installing a 
```
Say something like '*Many people use `git` through the commandline but this isn't everybody's cup of tea. Thankfully* you can do most git operations...

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+GitHub, we recommend `GitHub Desktop`_
+
+When working with the website on your workstation, you will want a good text 
+editor if you don't already have one, we recommend `Atom`_.
```
Add a clarification about text editors != WYSIWYG editors (like libreoffice).

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+Git on the Desktop
+~~~~~~~~~~~~~~~~~~
+
+There are a number of good Git desktop applications, but for working with 
```
Add '*Another alternative to using `git` on the command-line is a Git Desktop application.*' to the start of this paragraph.

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+go, you'll just need to insure that you have access to the OSL Pelican Git 
+repositories. See the `official Git web page`_ to install Git on your platform.
+
+The Github Desktop for Windows package comes with a Git Shell program for 
```
This might work better as a `.. note::` directive.

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+Getting Access
+--------------
+
+In order to make updates to the pelican sites, you will need access to the OSL 
```
Link to https://github.com/osuosl/ somewhere in here.

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+
+**Summary**
+
+#.	Check out the Git repository and create a new branch
```
You're mixing tabs and spaces here.

ElijahCaine commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+Editing Content.
+
+**Command Line**
+
```
A short preamble like '*Step zero is to clone the repository into your personal projects directory. Navigate to where you usually clone repositories and run the following command:* '

ElijahCaine commented:
@Kennric Sorry for the over-abundance of feedback but I think if you address the line notes I gave it'll be a really solid document. It is already very thorough and I'd be happy with it shipping the way it is, most of my feedback is just nit-picky or flow related.

ramereth commented on source/howtos/updating_pelican_sites.rst:
```diff
@@ -0,0 +1,601 @@
+~~~~~~~~~~~~~~~~~~~~~~~
+
+If you already are used to working with Git on a command line, you are good to 
+go, you'll just need to insure that you have access to the OSL Pelican Git 
```
s/insure/ensure/

ramereth commented:
@Kennric Overall I think this is a great doc. My only concern is how verbose it is and it doesn't do a walk through example. It might be good to make another page which is just a walk through example of just editing a page which is going to be the most common task people will do.

Kennric commented:
@ElijahCaine - thanks for all the input, I've made some changes, if you have a moment to check it out.

Kennric commented:
@ramereth I would like to do a walkthrough, I just haven't been able to put it together yet - I think I won't know how to do it well until we actually do a  training with editors. I'd like to keep this verbose as a reference (possibly split out commandline instructions) and create a walk-through separately.

Kennric commented:
@ramereth @ElijahCaine @kelnera 

Emergency merge request  - this document is going into service and needs to be on the wiki as of today. 

doc builds cleanly and appears to be reasonably comprehensive, new issues with it can be filed as bugs.

ramereth commented:
:+1:
